0001-customer-onboarding.md

```diff
@@ -0,0 +1,122 @@
+
+- **Namspace**. A new namespace in the Scipian cluster will be created for a
+given team, under a unique name tied to a specific project or application.
+- **Role**. This will define a role in Scipian as to what operations are allowed 
+in the cluster for anyone asigned to that role. It will be tied to the created 
+namespace, and only allow the least privileges required to create Workspaces and 
+Runs in Scipian.
+- **RoleBinding**. This will bind the created role to a set of users associated 
+with the onboarding team.
+
+If a team is already using Terraform to manage their infrastructure, Scipian
+supports the importing of Terraform state from a location external to Scipian.
+Scipian will pull this state and push it to it's own managed backend.
+
+Below is the flow when onboarding a new customer to Scipian:
+
+- New customer requests to be onboarded to Scipian
+- Customer provides list of users, (the email associated with the IDP Scipian 
+uses) of all the members who will need to use Scipian
+- A Scipian admin will create the Namespace and RBAC roles for the onboarding 
+team
+- Onboarding team members will use the Scipian Authentication CLI to set up 
+their kubeconfigs for interfacing Scipian.
+- The onboarding team will create a Kubernetes secret in their namespace with 
+the AWS credentials needed for Scipian to pull Terraform state from an S3 bucket
+- Scipian will pull that state into it's own backend
+- Customer will then be onboarded and ready to use the Scipian platform
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+- RBAC rules and Namespace will be created manually by a Scipian admin using
+yaml and `kubectl`. This could be automated later via an onboarding UI, but that
+is beyond the scope of this RFC.
+- Terraform state "puller" will be a Kubernetes job that runs a 
+special Docker container that will pull state from an S3 bucket,
+and place in the appropriate workspace in Scipian's backend. This could be
+initiated by a Scipian admin, or an onboarding user. The state puller will need
+to map the current Terraform state workspace structure created by `scipctl` to 
+the new workspace structure that Scipian expects.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This is a necessary feature of Scipian, and currently no drawbacks have been
+identified by adding this feature.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The onboarding outlined in this RFC will be an MVP level feature. Later, this
+functionality could be extended to include a self service UI, which would
+automate RBAC rules and Namespace creation, as well as the pulling of extneral
+Terraform state, thereby removing the requirement for a Scipian admin's direct
+involvement every time a new customer wishes to onboard to Scipian.
+
+# Prior art
+[prior-art]: #prior-art
+
+Because Scipian extends Kubernetes and utilizes many of Kubernetes' primitives,
```

Comment:
I haven't had time to really go through these two references yet, but I wanted to get them noted just to see if there's anything we should be looking into:

- [EPAM Cloud Terraform As A Service](https://cloud.epam.com/site/competency_center/e=p=c_services/terraform_as_a_service(=t=a=s))
- [Hashicorp Case Study - Terraform-Powered Platform-as-a-Service](https://www.hashicorp.com/resources/terraform-powered-platform-as-a-service) - video

Reply:
I'll take a look at these, but Scipian is a platform that extends much beyond infrastructure management (i.e. just Terraform). I haven't dug into these resources yet, but at a high level, these seem very focused on Terraform. Our goal is to iterate towards a place where we abstract Terraform away from the customer, and eventually even stop using it all together if it makes sense.